uniprotkb-rest/src/test/java/org/uniprot/api/uniprotkb/controller/UniProtKBByAccessionControllerIT.java

```diff
@@ -566,8 +573,103 @@ void searchWithIDForDeletedInactiveEntriesRedirectToAccession() throws Exception
+    }
+
+    @Test
+    void searchAccessionSpecificVersionRedirectToUnisave() throws Exception {
+        // when
+        ResultActions response =
+                getMockMvc()
+                        .perform(
+                                get(ACCESSION_RESOURCE, "A0A1J4H6S2")
+                                        .param("version", "5")
+                                        .header(ACCEPT, FF_MEDIA_TYPE_VALUE));
+        // then
+        response.andDo(log())
+                .andExpect(status().is(HttpStatus.SEE_OTHER.value()))
+                .andExpect(header().string(HttpHeaders.CONTENT_TYPE, FF_MEDIA_TYPE_VALUE))
+                .andExpect(
+                        header().string(
+                                HttpHeaders.LOCATION,
+                                getRedirectToEntryVersionPath("A0A1J4H6S2", "5", "txt")));
+    }
+
+    @Test
+    void searchAccessionSpecificVersionWithNonAllowedJsonFormat() throws Exception {
+        // when
+        ResultActions response =
+                getMockMvc()
+                        .perform(
+                                get(ACCESSION_RESOURCE, "A0A1J4H6S2")
+                                        .param("version", "last")
+                                        .header(ACCEPT, APPLICATION_JSON_VALUE));
+        // then
+        response.andDo(log())
+                .andExpect(status().is(HttpStatus.BAD_REQUEST.value()))
+                .andExpect(header().string(HttpHeaders.CONTENT_TYPE, APPLICATION_JSON_VALUE))
+                .andExpect(
+                        content()
+                                .string(
+                                        containsString(
+                                                "Expected one of [text/plain;format=fasta, text/plain;format=flatfile]")));
+    }
+
+    @Test
+    void searchAccessionSpecificVersionWithNonAllowedTsvFormat() throws Exception {
+        // when
+        ResultActions response =
+                getMockMvc()
+                        .perform(
+                                get(ACCESSION_RESOURCE, "A0A1J4H6S2")
+                                        .param("version", "5")
+                                        .header(ACCEPT, TSV_MEDIA_TYPE_VALUE));
+        // then
+        response.andDo(log())
+                .andExpect(status().is(HttpStatus.BAD_REQUEST.value()))
+                .andExpect(header().string(HttpHeaders.CONTENT_TYPE, TSV_MEDIA_TYPE_VALUE))
+                .andExpect(
+                        content()
+                                .string(
+                                        containsString(
+                                                "Expected one of [text/plain;format=fasta, text/plain;format=flatfile]")));
+    }
+
+    private String retrieveLastEntryVersion(String accession) throws Exception {
+        String entryVersionHistoryURI = "https://rest.uniprot.org/unisave/" + accession;
```

Comment:
Here in the test we are accessing external resources, this may cause issues and it is not recommended.